PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift

```diff
@@ -110,32 +105,34 @@ class SearchViewModel: ObservableObject {
     func clear() {
         searchResults = []
         subscriptions = []
+        self.savesLocalSearch.clear()
+        self.savesOnlineSearch.clear()
+        self.archiveOnlineSearch.clear()
+        self.allOnlineSearch.clear()
     }
 
-    func clearCache() {
-        cachedSearchResults = [
-            .saves: [:],
-            .archive: [:],
-            .all: [:]
-        ]
-    }
-
-    private func retrieveCachedResults() -> Bool {
-        if let cachedItems = cachedSearchResults[selectedScope]?[currentSearchTerm], !cachedItems.isEmpty {
-            self.searchResults = cachedItems
-            return true
-        }
-        return false
-    }
-
-    private func submitOnlineSearch(with term: String) {
-        clear()
-        searchService.results.receive(on: DispatchQueue.main).sink { [weak self] items in
-            guard let self else { return }
-            self.searchResults = items.compactMap { SearchItem(item: $0) }
-        }.store(in: &subscriptions)
-        Task {
-            await searchService.search(for: term, scope: selectedScope)
+    private func submitSearch(with term: String) {
+        switch selectedScope {
+        case .saves:
+            // TODO: Handle Offline for Premium https://getpocket.atlassian.net/browse/IN-971
+            guard isPremium else {
+                self.searchResults = savesLocalSearch.search(with: term)
```

Comment:
By setting all these to `self.searchResults` we still get the same issue where one search may be updating the value for a different scope. However, I am unable to see this visually as we get the correct caching now for the scopes. Do we prefer to separate out `self.searchResults` to `self.searchResultsForSaves`, `self.searchResultsForArchive`, `self.searchResultsForAll`?

This a small case I feel and wanted to double check with the team on the overhead of adding a results property for each scope vs recycling.